platform/http_request.cpp

```diff
@@ -386,6 +387,12 @@ HttpRequest::~HttpRequest()
 {
 }
 
+HttpRequest::ProgressT const & HttpRequest::Progress() const
+{
+  ASSERT(m_progress.first <= m_progress.second, ());
```

Comment:
Ассерт будет падать при штатной работе с загрузчиком. Я не понимаю, как можно исправить ситуацию, когда качается файл неизвестного заранее размера, но его прогресс загрузки, очевидно, идет - ведь это вполне штатный случай для HTTP.

-1 являлось признаком того, что размер файла не известен. Ноль - не является таким признаком, потому что размер файла на сервере может быть ноль, или размер ответа от сервера тоже может быть с BODY нулевой длины.

> On Apr 6, 2016, at 09:24, Илья Гречухин <notifications@github.com> wrote:
> 
> In platform/http_request.cpp <https://github.com/mapsme/omim/pull/2756#discussion_r58656420>:
> 
> > @@ -386,6 +387,12 @@ HttpRequest::~HttpRequest()
> >  {
> >  }
> >  
> > +HttpRequest::ProgressT const & HttpRequest::Progress() const
> > +{
> > +  ASSERT(m_progress.first <= m_progress.second, ());
> m_progress.first > m_progress.second противоречит определению типа HttpRequest::ProgressT
> Если будет возникать ошибка, нужно будет править то место откуда она возникнет.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2756/files/193bf11a2e6f8a3ead0d909e7f91c3f351ef2e0a#r58656420>